src/hooks/use-dhis2-deployment-update.ts

```diff
@@ -0,0 +1,54 @@
+import { FORM_ERROR } from 'final-form'
+import { useCallback } from 'react'
+import type { AnyObject } from 'react-final-form'
+import { useNavigate } from 'react-router'
+import { UpdateDeploymentRequest, Deployment } from '../types/index.ts'
+import { useAuthAxios } from './use-auth-axios.ts'
+import { useDhis2InstanceUpdate } from './use-dhis2-instance-update.ts'
+
+export const useDhis2DeploymentUpdate = (deploymentId: number, deployment?: Deployment) => {
+    const navigate = useNavigate()
+    const [, executePut] = useAuthAxios(
+        {
+            url: `/deployments/${deploymentId}`,
+            method: 'PUT',
+        },
+        { manual: true }
+    )
+    const updateInstance = useDhis2InstanceUpdate(deploymentId)
+
+    const updateDeployment = useCallback(
+        async (values: AnyObject) => {
+            try {
+                const payload: UpdateDeploymentRequest = {
+                    description: values.description,
+                    ttl: values.ttl,
+                }
+                await executePut({ data: payload })
+
+                if (deployment?.instances) {
+                    for (const instance of deployment.instances) {
+                        const stackParams = values[instance.stackName]
+                        if (stackParams) {
+                            await updateInstance({
+                                instanceId: instance.id,
+                                stackName: instance.stackName,
+                                parameters: stackParams,
+                                publicValue: instance.stackName === 'dhis2-core' ? values.public : undefined,
+                            })
+                        }
+                    }
+                }
```

Comment:
With this we'll trigger an update of all Instances within a Deployment, even if only one of the Instances' parameters are changed, right? If we only update `dhis2-core` params, `dhis2-db` shouldn't be updated.